copier/user_data.py

```diff
@@ -395,6 +406,24 @@ def render_value(self, value: Any) -> str:
         except UndefinedError as error:
             raise UserMessageError(str(error)) from error
 
+    def render_validation(self, template_string: str, value: str) -> str:
+        """Render a single jinja validation template value.
+
+        Validation templates will return an empty string on success and a
+        non-empty string as the error message. The error message defined in a
+        template might contain indents, which will be stripped from the returned
+        error message.
+        """
+        template = self.jinja_env.from_string(template_string)
+        answers = {**self.answers.combined, **{self.var_name: value}}
```

Comment:
Instead of adding this whole new method, just add the `**{self.var_name: value}` part to [the call to `template.render` inside `render_value()`](https://github.com/copier-org/copier/pull/698/files#diff-520654b89873468fcdb6a0317f022436ab8f3bc467e56ea2d7cc0102c2ed4b00L394). That should be enough.